Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -52,9 +52,8 @@ private sealed class ReloadModulesQueueItem : IAnalyzable {
                 }
 
                 var currentTcs = Interlocked.Exchange(ref _tcs, new TaskCompletionSource<bool>());
-                var task = Task.Run(() => _analyzer.ReloadModulesAsync(), cancel);
                 try {
-                    task.WaitAndUnwrapExceptions();
+                    _analyzer.ReloadModulesAsync().WaitAndUnwrapExceptions();
```

Comment:
The docs for `ReloadModulesAsync` explicitly say it needs to be run on the analyzer thread. The only async operation within this function (currently) is run on a separate thread, so there shouldn't be any risk of deadlock here.